### PR TITLE
Add msys2 resource

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
@tyler-ball @btm

File: resources/msys2_package.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  def msys2_init
+    cache_dir = ::File.join(root, '.cache')
+    f_cache_dir = win_friendly_path(cache_dir)
+    url = 'http://iweb.dl.sourceforge.net/project/msys2/Base/x86_64/msys2-base-x86_64-20160205.tar.xz'
+
+    unless ::File.exist?(::File.join(root, 'msys2.exe'))
+      seven_zip_archive "cache msys2 base to #{f_cache_dir}" do
+        source url
+        path f_cache_dir
+        checksum '7e97e2af042e1b6f62cf0298fe84839014ef3d4a3e7825cffc6931c66cc0fc20'
```
Comment (btm) [inline resources/msys2_package.rb]: put this checksum up with the url since they're related?

**Comment by btm** [comment]
it feels strange having the msys2_init stuff in the package resource, but I guess that'll get refactored whenever we need it elsewhere and is probably the expedient choice for now. :+1:

**Comment by ksubrama** [comment]
Oh I despise having to do the init stuff inside the msys2 package resource.  But every other cookbook seems to assume that the resource its manipulating actually exists.  rpm_package doesn't install RHEL or Fedora.  powershell_script doesn't install WMF.

Would it be better if I moved this init stuff into a recipe that installed msys2 once?  The danger with putting anything into a recipe is that oh god how do we hide to details of our implementation in case we need to refactor it later.  I don't particularly care either way.  I think even the https://github.com/chocolatey/chocolatey-cookbook/blob/master/recipes/default.rb Chocolatey cookbook assumes you're going to install chocolate first before installing packages using it.

**Comment by tyler-ball** [comment]
From a user point of view I like that I don't have to think about installing msys2. I just use the `msys2_package` resource and it takes care of installing the backend if necessary. It doesn't fit the existing chocolatey cookbook pattern, but it seems nicer. See any issue with that @btm?

Now it just needs some tests and documentation :)

**Comment by btm** [comment]
It's acceptable as is. There could be a compromise in http://www.fewbytes.com/using-include_recipe-in-chef-lwrp/, but that's a little hairy.

**Comment by ksubrama** [comment]
Added documentation and some basic integration tests.  This should now be ready.

**Pull request merged**